astroid/nodes/scoped_nodes.py

```diff
@@ -637,12 +652,36 @@ def fully_defined(self):
         """
         return self.file is not None and self.file.endswith(".py")
 
-    def statement(self):
+    @overload
+    def statement(self, *, future: Literal[None] = ...) -> "Module":
+        ...
+
+    # pylint: disable-next=signature-differs
+    # https://github.com/PyCQA/pylint/issues/5264
```

Comment:
```suggestion
```
It seems like your luck didn't outran us yet. This actual now triggers a `useless-suppression` message. Guess we found a `false-negative` by adding `*` 😅 https://github.com/PyCQA/pylint/issues/5270

Reply:
Is `useless-suppression` not enabled for `astroid`? CI is passing.

We might want to enable that in another PR.

Reply:
It's enabled, but not a failure condition! I don't think we should change that.

Reply:
Sure? Seems like something that will be easily missed in review

Reply:
I also think we should make it a failure condition so it's handled in the proper MR each time.

Reply:
> We might want to enable that in another PR.

It would be as easy as adding `--fail-on=useless-suppression` to the `pylint` command.

The issue is that some errors might be Python version specific. It could create a situation where only one, `pre-commit` or `CI`, will succeed when tested with different versions. The `py-version` option should help, but I don't know if it's enough.

Additionally, this would prevent us from having the latest `pylint` version installed in a local env as this is likely to have fixed some false-positives. -> Which would then trigger `useless-suppression`.

Reply:
That last point will indeed be difficult to avoid. I'll try and think of something to fix that, but in the meantime let's let this rest!

Reply:
Do you maybe want to track this as issue in `pylint`?

> in the meantime let's let this rest!

I completely agree!

Reply:
Shouldn't this be tracked in `astroid`? `pylint` has its own issues with `useless-suppression` (https://github.com/PyCQA/pylint/issues/5040) 😄

Reply:
😂 Though if we enable it, we'll likely want to do it for `pylint` too. Tbh I don't really care about it due to the issues mentioned earlier.